modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -1322,22 +1321,12 @@ private CompletableFuture<List<BinaryRow>> scanSortedIndex(
 
         int flags = request.flags();
 
-        Supplier<ManuallyCloseable> cursorProvider = ((SortedIndexStorage) schemaAwareIndexStorage.storage()).readOnlyScanImplemented()
-                ? () -> new CursorResource(
-                indexStorage.readOnlyScan(
+        Cursor<IndexRow> cursor = remotelyTriggeredResourceRegistry.<CursorResource>register(cursorId, request.coordinatorId(),
+                () -> new CursorResource(indexStorage.scan(
```

Comment:
This one should be a readonly scan, now readwrite scan

Reply:
lost a change( fixed